Handler/Demo.hs

```diff
@@ -0,0 +1,35 @@
+module Handler.Demo where
+
+import Import
+import Model.Site
+
+import Data.Maybe (fromJust, listToMaybe)
+
+getDemoR :: Handler Html
+getDemoR = do
+    root <- getAppRoot
+    siteId <- runDB $ findDemoSite root
+
+    defaultLayout $ do
+        setTitle "Carnival Demo"
+        $(widgetFile "demo")
+
+findDemoSite :: Text -> DB SiteId
+findDemoSite root = do
+    admin <- getAdmin
+    msite <- listToMaybe <$> findSites admin [SiteBaseUrl ==. root]
+
+    entityKey <$> maybe (createDemoSite admin root) return msite
+
+createDemoSite :: UserId -> Text -> DB (Entity Site)
+createDemoSite userId root = createSite userId Site
+    { siteName = "carnival-demo"
+    , siteBaseUrl = root
+    , siteRssAuthor = "Carnival Demo"
+    , siteRssTitle = "Comments on Carnival Demo"
+    , siteRssDescription = "Comments on Carnival Demo"
+    , siteRssLanguage = "en-us"
+    }
+
+getAdmin :: DB UserId
+getAdmin = entityKey . fromJust <$> getBy (UniqueUser "github" "50812")
```

Comment:
Will this always exist, or do we need to create it as above?

Seems kind of strange to embed this ID in here.